pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -623,29 +646,33 @@ func constructFilesSecret(driver string, config map[string]interface{}) (*corev1
 	return nil, nil
 }
 
-func shouldForceRemoval(job *batchv1.Job, d data.Object) bool {
-	var createJobCondition summary.Condition
-	for _, c := range summary.GetUnstructuredConditions(d) {
-		if c.Type() == createJobConditionType {
-			createJobCondition = c
-			break
-		}
+func shouldCleanupObjects(job *batchv1.Job, d data.Object) bool {
+	if !job.Status.CompletionTime.IsZero() {
+		return true
+	}
+
+	createJobCondition := getCondition(d, createJobConditionType)
+	if createJobCondition == nil {
+		return false
 	}
 
 	forceRemoveAnnValue, _ := d.Map("metadata", "annotations")[forceRemoveMachineAnn].(string)
 
-	if createJobCondition.Reason() != string(errors.CreateMachineError) &&
-		strings.ToLower(forceRemoveAnnValue) != "true" {
-		return false
+	if createJobCondition.Reason() == string(errors.CreateMachineError) || strings.ToLower(forceRemoveAnnValue) == "true" {
+		return strings.ToLower(job.Spec.Template.Labels[InfraJobRemove]) == "true" && condition.Cond("Failed").IsTrue(job)
 	}
 
-	return condition.Cond("Failed").IsTrue(job)
+	return false
 }
 
 func FailedMachineDeleteMessage(kind, failureReason, failureMessage string) string {
 	return fmt.Sprintf("failed deleting server (%s) in infrastructure provider: %s: %s", kind, failureReason, failureMessage)
 }
 
+func FailedMachineCreateMessage(kind, failureReason, failureMessage string) string {
+	return fmt.Sprintf("failed creatinging server (%s) in infrastructure provider: %s: %s", kind, failureReason, failureMessage)
```

Comment:
A bit too many `ing`s :)
```suggestion
	return fmt.Sprintf("failed creating server (%s) in infrastructure provider: %s: %s", kind, failureReason, failureMessage)
```

Reply:
Fixed.